ecl/hql/hqlutil.cpp

```diff
@@ -10212,6 +10212,15 @@ IHqlExpression * convertSetToExpression(bool isAll, size32_t len, const void * p
                 presult += numUChars;
             };
             break;
+        case type_utf8:
+            while (presult < presult_end)
+            {
+                const size32_t numUChars = *((size32_t *) presult);
+                presult += sizeof(size32_t);
+                results.append(*createConstant(createUtf8Value((unsigned)numUChars, (const char*)presult, makeUtf8Type(numUChars, NULL))));
+                presult += numUChars;
```

Comment:
I think this should be.

presult += rtlUtf8Size(numUChars, presult)

Is there a test case where the set returns multi-byte utf-8 characters?

Reply:
Also, added test cases for multibyte Unicode and UTF-8 REGEXFINDSET.  This uncovered some issues in both folded and unfolded code, cascading some further changes that will be in the next commit.